content/200-orm/200-prisma-client/500-deployment/301-edge/450-deploy-to-cloudflare.mdx

```diff
@@ -536,6 +545,193 @@ npx wrangler deploy
 
 The command will output the URL where you can access the deployed Worker.
 
-### D1
+### Cloudflare D1
+
+If you are using a D1 database, you need to:
+
+- use the `@prisma/adapter-d1` database adapter (via the `driverAdapters` Preview feature)
+- set `sqlite` as the `datasource` provider in your Prisma schema
+- manually generate SQL statements for schema changes using `prisma migrate diff` but execute them using [D1's migration system](https://developers.cloudflare.com/d1/reference/migrations/)
+
+You can find a [deployment-ready example on GitHub](https://github.com/prisma/prisma-examples/blob/latest/deployment-platforms/edge/cloudflare-workers/with-d1).
+
+#### 1. Configure Prisma schema
+
+> **Note**: If you don't have a project to deploy, follow the instructions in the [Prerequisites](#prerequisites) to bootstrap a basic Cloudflare Worker with Prisma ORM in it.
+
+In your Prisma schema, add the `driverAdapters` Preview feature to the `generator` block and set the `provider` of the `datasource` to `sqlite`. If you just bootstrapped the Prisma schema with `prisma init`, also be sure to add the following `User` model to it:
+
+```prisma file=schema.prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["driverAdapters"]
+}
+
+datasource db {
+  provider = "sqlite"
+  url      = env("DATABASE_URL")
+}
+
+model User {
+  id    Int     @id @default(autoincrement())
+  email String  @unique
+  name  String?
+}
+```
+
+Note that in this tutorial, you won't need the `.env` file since the connection between Prisma ORM and D1 will happen through a [binding](https://developers.cloudflare.com/workers/configuration/bindings/).
+
+#### 2. Install dependencies
+
+Next, install the required packages:
+
+```terminal
+npm install @prisma/adapter-d1
+```
```

Comment:
I'd add a recommendation to use the latest version of Wrangler (e.g., `wrangler@^3.39.0`), otherwise the `--remote` flag in `wrangler d1 migrations apply <DATABASE_NAME> --remote` won't be available.

Reply:
ha good callout! I added a note about this here 👍